ProcessLib/SmallDeformationWithLIE/Common/PostUtils.cpp

```diff
@@ -0,0 +1,205 @@
+    std::vector<MeshLib::Node*> new_nodes(MeshLib::copyNodeVector(org_mesh.getNodes()));
+    std::vector<MeshLib::Element*> new_eles(
+        MeshLib::copyElementVector(org_mesh.getElements(), new_nodes));
+
+    // duplicate fracture nodes
+    for (auto const* org_node : vec_fracture_nodes)
+    {
+        auto duplicated_node = new MeshLib::Node(org_node->getCoords(), new_nodes.size());
+        new_nodes.push_back(duplicated_node);
+        _map_dup_newNodeIDs[org_node->getID()] = duplicated_node->getID();
+    }
+
+    // split elements using the new duplicated nodes
+    auto prop_levelset = org_mesh.getProperties().getPropertyVector<double>("levelset1");
+    for (auto const* org_e : vec_fracutre_matrix_elements)
+    {
+        // only matrix elements
+        if (org_e->getDimension() != org_mesh.getDimension())
+            continue;
+
+        auto const eid = org_e->getID();
+
+        // keep original if the element has levelset=0
+        if ((*prop_levelset)[eid] == 0)
+            continue;
+
+        // replace fracture nodes with duplicated ones
+        MeshLib::Element* e = new_eles[eid];
+        for (unsigned i=0; i<e->getNumberOfNodes(); i++)
+        {
+            auto itr = _map_dup_newNodeIDs.find(e->getNodeIndex(i));
+            if (itr == _map_dup_newNodeIDs.end())
+                continue;
+
+            auto itr2 = std::find_if(vec_fracture_nodes.begin(), vec_fracture_nodes.end(),
+                                     [&](MeshLib::Node const*node) { return node->getID()==e->getNodeIndex(i);});
+            if (itr2 == vec_fracture_nodes.end())
+                continue;
+            e->setNode(i, new_nodes[itr->second]);
+        }
+    }
+
+    // new mesh
+    _output_mesh.reset(new MeshLib::Mesh(org_mesh.getName(), new_nodes, new_eles));
+    createProperties<int>();
+    createProperties<double>();
+    copyProperties<int>();
+    copyProperties<double>();
+    calculateTotalDisplacement();
+}
+
+template <typename T>
+void PostProcessTool::createProperties()
+{
+    MeshLib::Properties const& src_properteis = _org_mesh.getProperties();
+    for (auto name : src_properteis.getPropertyVectorNames())
+    {
+        auto const* src_prop = src_properteis.getPropertyVector<T>(name);
+        if (!src_prop)
+            continue;
+        auto const n_src_comp = src_prop->getNumberOfComponents();
+        // convert 2D vector to 3D. Otherwise Paraview Calculator filter does not recognize
+        // it as a vector
+        auto const n_dest_comp = (n_src_comp==2) ? 3 : n_src_comp;
+
+        if (src_prop->getMeshItemType() == MeshLib::MeshItemType::Node)
+        {
+            auto new_prop =
+                _output_mesh->getProperties().createNewPropertyVector<T>(
+                    name, MeshLib::MeshItemType::Node, n_dest_comp);
+            new_prop->resize(_output_mesh->getNumberOfNodes() * n_dest_comp);
+        }
+        else if (src_prop->getMeshItemType() == MeshLib::MeshItemType::Cell)
```

Comment:
If we later on add some field properties to VTK files, we will have to extend this code. We should keep that in mind.